portal/views/reporting.py

```diff
@@ -284,25 +290,38 @@ def questionnaire_status():
             historic['status'] = status
             historic['qb'] = qbd.questionnaire_bank.name
             historic['visit'] = visit_name(qbd)
+            entry_method = QNR_results(
+                patient, qbd.qb_id, qbd.iteration).entry_method()
+            if entry_method:
+                historic['entry_method'] = entry_method
+            else:
+                historic.pop('entry_method', None)
             results.append(historic)
 
     if request.args.get('format', 'json').lower() == 'csv':
         def gen(items):
-            desired_order = [
-                'user_id', 'study_id', 'status', 'visit', 'site', 'consent']
-            yield ','.join(desired_order) + '\n'  # header row
+            desired_order = (
+                'user_id', 'study_id', 'status', 'visit',
+                'entry_method', 'site', 'consent')
+            yield desired_order  # header row
             for i in items:
-                yield ','.join(
-                    [str(i.get(k, "")) for k in desired_order]) + '\n'
+                if int(i['user_id']) not in (1707, 1726):
+                    continue
+                yield [str(i.get(k, "")) for k in desired_order]
         # default file base title
         base_name = 'Questionnaire-Timeline-Data'
         if org_id:
             base_name = '{}-{}'.format(
                 base_name,
                 Organization.query.get(org_id).name.replace(' ', '-'))
         filename = '{}-{}.csv'.format(base_name, strftime('%Y_%m_%d-%H_%M'))
+
+        si = StringIO()
+        cw = csv.writer(si)
+        cw.writerows(gen(results))
+
         return Response(
-            gen(results),
+            si.getvalue(),
```

Comment:
I think this no [longer streams the CSV response](http://flask.pocoo.org/docs/1.0/patterns/streaming/), which may be desirable for enormous datasets.

Using the `csv` module would be a great improvement for standardization if we could get it to work in the same way...

Reply:
Found [a solution|https://stackoverflow.com/a/32609533] to bring back streaming using the csv module.